File: application/app/Http/Controllers/Api/UserController.php
<?php

namespace App\Http\Controllers\Api;

use App\Http\Requests\UpdateDarkModeRequest;
use Illuminate\Http\Request;

use App\Services\UserService;

class UserController extends Controller
{
    private UserService $service;

    public function __construct(UserService $userService)
    {
        $this->service = $userService;
    }

    public function getUserById(Request $request): array
    {
        try {
            return $this->service->getUserById($request->user()->id);
        } catch (\Exception $e) {
            throw new \Exception("Failed to get user");
        }
    }

    public function updateDarkMode(UpdateDarkModeRequest $request): bool
    {
        try {
            return $this->service->updateDarkMode(['userId' => $request->user()->id, 'darkMode' => $request->input('darkMode')]);
        } catch(\Exception $e) {
            throw new \Exception('Failed to update dark mode', 500);
        }
    }
}

File: server/BuilderApi/Controllers/UserController.cs
using BuilderServices.UserService;
using BuilderServices.UserService.Requests;
using BuilderServices.UserService.Responses;
using Microsoft.AspNetCore.Mvc;

namespace BuilderApi.Controllers;

[ApiController]
[Route("api/user")]
public class UserController(
    UserService service
) : ControllerBase
{
    [HttpGet("")]
    public async Task<IActionResult> GetUser()
    {
        var user = await service.GetLimitedUserByIdAsync().ConfigureAwait(false);

        return Ok(user);
    }

    [HttpPatch("update/settings/darkMode")]
    public async Task<IActionResult> UpdateDarkMode([FromBody] UpdateDarkModeRequest request)
    {
        await service.UpdateDarkModeAsync(request.IsDarkMode).ConfigureAwait(false);

        return Ok(new UpdateDarkModeResponse
        {
            IsUpdated = true
        });
    }
}
